libdnf/conf/vars.cpp

```diff
@@ -204,21 +205,53 @@ std::string Vars::substitute(const std::string & text) const {
     return res;
 }
 
-void Vars::load(const std::string & installroot, const std::vector<std::string> & directories) {
-    detect_vars(installroot);
+void Vars::set(const std::string & name, const std::string & value, Priority prio) {
+    auto it = variables.find(name);
+    if (it != variables.end()) {
+        if (it->second.priority > prio) {
+            return;
+        }
+        it->second.value = value;
+        it->second.priority = prio;
+        return;
+    }
+    variables.insert({name, {value, prio}});
+}
 
+void Vars::load(const std::string & installroot, const std::vector<std::string> & directories) {
     load_from_env();
 
     for (const auto & dir : directories) {
         load_from_dir(std::filesystem::path(installroot) / dir);
     }
+
+    detect_vars(installroot);
 }
 
 void Vars::detect_vars(const std::string & installroot) {
-    auto arch = detect_arch();
-    values["arch"] = arch;
-    values["basearch"] = get_base_arch(arch.c_str());
-    values["releasever"] = detect_release(installroot);
+    auto it = variables.find("arch");
+    if (it == variables.end()) {
+        variables.insert({"arch", {detect_arch(), Priority::AUTO}});
+    } else if (it->second.priority <= Priority::AUTO) {
+        it->second.value = detect_arch();
+        it->second.priority = Priority::AUTO;
+    }
+
+    it = variables.find("basearch");
+    if (it == variables.end()) {
+        variables.insert({"basearch", {get_base_arch(variables["arch"].value.c_str()), Priority::AUTO}});
```

Comment:
`get_base_arch()` can return nullptr, therefore it can result in crash. What about to test it and throw an exception when unsupported `basearch` appears.